packages/compiler-sfc/src/compileScript.ts

```diff
@@ -101,7 +101,8 @@ export function compileScript(
     sfc.template && sfc.template.attrs['inherit-attrs'] === 'false'
   const scriptLang = script && script.lang
   const scriptSetupLang = scriptSetup && scriptSetup.lang
-  const isTS = scriptLang === 'ts' || scriptSetupLang === 'ts'
+  const isTS =
+    scriptLang === 'ts' || scriptLang === 'tsx' || scriptSetupLang === 'ts'
```

Comment:
Does`setup` syntax support `tsx` ?

Reply:
We almost never write j/tsx in `<script setup>`, so that's fine.

Reply:
I think we **can** support tsx in `<script setup>` by adding `tsx` in [compileScript.ts#L159](https://github.com/vuejs/vue-next/blob/870f2a7ba35245fd8c008d2ff666ea130a7e4704/packages/compiler-sfc/src/compileScript.ts#L159). Any reason for not adding it here? 🤔

Reply:
Explained above

Reply:
> We almost never write j/tsx in `<script setup>`

I don't think it's a good reason for leaving `tsx` broken in `setup` script though. 😕 

Anyway, should I open a new PR to enable `tsx` in `setup` syntax or can we just enable it here?

Reply:
There's little reason to write j/tsx in `<script setup>`, but yeah technical people *can* if they really want to, so it makes sense to correctly handle it. (#3825)